analyzers/rspec/cs/S2445_c#.html

```diff
@@ -0,0 +1,44 @@
+<p>Locking on a class field synchronizes not on the field itself, but on the object assigned to it. So locking on a non-<code>readonly</code> field
+makes it possible for the field’s value to change while a thread is in a block locked on the old value. That would allow a second thread, locked on
+the new value, to enter the block at the same time.</p>
+<p>The story is very similar for locking on a local variable or on a new instance; two different threads running the method in parallel would lock on
```

Comment:
For the `local variable` part, check https://github.com/SonarSource/sonar-dotnet/pull/6750/files#r1106924850